File: binary_search/0392_is_subsequence/0392_is_subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        if not s :
            return True
        
        i = 0
        for c in t:
            if c == s[i]:
                i += 1
            if i == len(s):
                break
        return i == len(s)

s = "abc"
t = "ahbgdc"
res = Solution().isSubsequence(s,t)
print(res)

File: binary_search/0392_is_subsequence/0392_is_subsequence.go
package main 

import "fmt"

func isSubsequence(s, t string) bool {
	for len(s) > 0 && len(t) > 0 {
		if s[0] == t[0] {
			s = s[1:]
		}
		t = t[1:]
	}
	return len(s) == 0
}

func main() {
	s := "abc"
	t := "ahbgdc"
	res := isSubsequence(s,t)
	fmt.Println(res)
}